src/main/java/chopchop/logic/commands/MakeRecipeCommand.java

```diff
@@ -0,0 +1,100 @@
+package chopchop.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import chopchop.commons.core.Messages;
+import chopchop.logic.commands.exceptions.CommandException;
+import chopchop.logic.parser.ItemReference;
+import chopchop.model.Model;
+import chopchop.model.exceptions.IncompatibleIngredientsException;
+import chopchop.model.ingredient.Ingredient;
+import chopchop.model.ingredient.IngredientReference;
+import chopchop.model.recipe.Recipe;
+import chopchop.util.Pair;
+
+/**
+ * Makes a dish according to the recipe identified by the index number or name used in the displayed recipe list,
+ * removing the ingredients used.
+ */
+public class MakeRecipeCommand extends Command {
+    public static final String COMMAND_WORD = "make recipe";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Makes a dish according to the recipe identified by the index number or name used in the displayed "
+            + "recipe list, removing the ingredients used.\n"
+            + "Parameters: INDEX (must be a positive integer) / NAME\n"
+            + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_MAKE_RECIPE_SUCCESS = "Recipe made: %s";
+    public static final String MESSAGE_MAKE_RECIPE_ERROR = "Cannot make recipe due to ingredient '%s'";
+    public static final String MESSAGE_RECIPE_NOT_FOUND = "No recipe named '%s'";
+    public static final String MESSAGE_INGREDIENT_NOT_FOUND = "No ingredient named '%s'";
+
+    private final ItemReference item;
+    // (Original ingredient, (Consumed ingredient, Remaining ingredient))
+    private List<Pair<Ingredient, Pair<Ingredient, Ingredient>>> ingredients;
```

Comment:
maybe you should make a smol nested helper class for this

Reply:
Was lazy, will do.